controlplane/rosa/controllers/rosacontrolplane_controller.go

```diff
@@ -0,0 +1,445 @@
+	}
+	stsBuilder = stsBuilder.OperatorIAMRoles(roles...)
+
+	instanceIAMRolesBuilder := cmv1.NewInstanceIAMRoles()
+	instanceIAMRolesBuilder.MasterRoleARN("TODO")
+	instanceIAMRolesBuilder.WorkerRoleARN("TODO")
+	stsBuilder = stsBuilder.InstanceIAMRoles(instanceIAMRolesBuilder)
+	stsBuilder.OidcConfig(cmv1.NewOidcConfig().ID(*rosaScope.ControlPlane.Spec.OIDCID))
+
+	stsBuilder.AutoMode(true)
+
+	awsBuilder := cmv1.NewAWS().
+		AccountID(*rosaScope.ControlPlane.Spec.AccountID)
+	awsBuilder = awsBuilder.SubnetIDs(rosaScope.ControlPlane.Spec.Subnets...)
+	awsBuilder = awsBuilder.STS(stsBuilder)
+	clusterBuilder = clusterBuilder.AWS(awsBuilder)
+
+	clusterNodesBuilder := cmv1.NewClusterNodes()
+	clusterNodesBuilder = clusterNodesBuilder.AvailabilityZones(rosaScope.ControlPlane.Spec.AvailabilityZones...)
+	clusterBuilder = clusterBuilder.Nodes(clusterNodesBuilder)
+
+	clusterProperties := map[string]string{}
+	clusterProperties[rosaCreatorArnProperty] = *rosaScope.ControlPlane.Spec.CreatorARN
+
+	clusterBuilder = clusterBuilder.Properties(clusterProperties)
+	clusterSpec, err := clusterBuilder.Build()
+	if err != nil {
+		return ctrl.Result{}, fmt.Errorf("failed to create description of cluster: %v", err)
+	}
+
+	// Create a logger that has the debug level enabled:
+	ocmLogger, err := sdk.NewGoLoggerBuilder().
+		Debug(true).
+		Build()
+	if err != nil {
+		return ctrl.Result{}, fmt.Errorf("failed to build logger: %w", err)
+	}
+
+	// Create the connection, and remember to close it:
+	token := os.Getenv("OCM_TOKEN")
+	connection, err := sdk.NewConnectionBuilder().
+		Logger(ocmLogger).
+		Tokens(token).
+		URL(ocmAPIUrl).
+		Build()
+	if err != nil {
+		return ctrl.Result{}, fmt.Errorf("failed to build connection: %w", err)
+	}
+	defer connection.Close()
+
+	log := logger.FromContext(ctx)
+	cluster, err := connection.ClustersMgmt().V1().Clusters().
+		Add().
+		// Parameter("dryRun", *config.DryRun).
+		Body(clusterSpec).
+		Send()
+	if err != nil {
+		log.Info("error", "error", err)
+		return ctrl.Result{RequeueAfter: 10 * time.Second}, nil
+	}
+
+	clusterObject := cluster.Body()
+	log.Info("result", "body", clusterObject)
+
+	return ctrl.Result{}, nil
+}
+
+func (r *ROSAControlPlaneReconciler) reconcileDelete(_ context.Context, rosaScope *scope.ROSAControlPlaneScope) (ctrl.Result, error) {
+	// log := logger.FromContext(ctx)
+
+	rosaScope.Info("Reconciling ROSAControlPlane delete")
+
+	// Create a logger that has the debug level enabled:
+	ocmLogger, err := sdk.NewGoLoggerBuilder().
+		Debug(true).
+		Build()
+	if err != nil {
+		return ctrl.Result{}, fmt.Errorf("failed to build logger: %w", err)
+	}
+
+	// Create the connection, and remember to close it:
+	token := os.Getenv("OCM_TOKEN")
+	connection, err := sdk.NewConnectionBuilder().
+		Logger(ocmLogger).
+		Tokens(token).
+		URL(ocmAPIUrl).
+		Build()
+	if err != nil {
+		return ctrl.Result{}, fmt.Errorf("failed to build connection: %w", err)
+	}
+	defer connection.Close()
```

Comment:
I'd prefer not to ignore this error. You could do it like this:

```go
func (r *ROSAControlPlaneReconciler) reconcileDelete(_ context.Context, rosaScope *scope.ROSAControlPlaneScope) (_ ctrl.Result, err error) {
```

And then:

```go
	defer func() {
		if cerr := connection.Close(); cerr != nil {
			err = errors.Join(err, cerr)
		}
	}()
```

Nothing else should be changed.

Reply:
Same for the normal flow defer close.

Reply:
thanks, done